### PR TITLE
NIAD-380: Acceptance date mapper and changes to DateTimePeriod segmen…

### PR DESCRIPTION
**Pull request by mattd-kainos**
…t needed for all acceptance transactions

### CODE REVIEW

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/mapper/AcceptanceDateMapper.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+public class AcceptanceDateMapper implements FromFhirToEdifactMapper<DateTimePeriod> {
+
+    public DateTimePeriod map(Parameters parameters) {
+        return new DateTimePeriod(getAcceptanceDate(parameters), DateTimePeriod.TypeAndFormat.ACCEPTANCE_DATE);
+    }
+
+    private Instant getAcceptanceDate(Parameters parameters) {
+        String dateAsString = ParametersExtension.extractValue(parameters, ParameterNames.ACCEPTANCE_DATE);
+        return LocalDate.parse(dateAsString).atStartOfDay(ZoneOffset.UTC).toInstant();
```
Comment (bartek-sarul) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/mapper/AcceptanceDateMapper.java:22]: Shouldn't be UK time zone?

File: src/test/java/uk/nhs/digital/nhsconnect/nhais/mapper/AcceptanceDateMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            .setName(ParameterNames.ACCEPTANCE_DATE)
+            .setValue(new StringType("2020-06-05"));
+
+        var acceptanceDateMapper = new AcceptanceDateMapper();
+        DateTimePeriod dateTimePeriod = acceptanceDateMapper.map(parameters);
+
+        var expectedInstant = LocalDate.parse("2020-06-05").atStartOfDay(ZoneOffset.UTC).toInstant();
+        var expectedDateTimePeriod = new DateTimePeriod(expectedInstant, DateTimePeriod.TypeAndFormat.ACCEPTANCE_DATE);
+
+        assertEquals(expectedDateTimePeriod, dateTimePeriod);
```
Comment (bartek-sarul) [inline src/test/java/uk/nhs/digital/nhsconnect/nhais/mapper/AcceptanceDateMapperTest.java:33]: We agreed to use assertj

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/mapper/AcceptanceDateMapper.java (hunk 1 above)
Comment (jaroslaw-wojciechowski) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/mapper/AcceptanceDateMapper.java:22]: not Europe/London ?

Comment (mattd-kainos) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/mapper/AcceptanceDateMapper.java:22]: From FHIR so ISO 8601 date. There are no time zones associated with dates.

Comment (mattd-kainos) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/mapper/AcceptanceDateMapper.java:22]: see other comment

File: src/test/java/uk/nhs/digital/nhsconnect/nhais/mapper/AcceptanceDateMapperTest.java (hunk 2 above)
Comment (mattd-kainos) [reply src/test/java/uk/nhs/digital/nhsconnect/nhais/mapper/AcceptanceDateMapperTest.java:33]: clone of adjacent class so consistent. I'll change if I have time.

**Review by bartek-sarul — APPROVED** [review]

**Review by jaroslaw-wojciechowski — APPROVED** [review]

**Pull request merged**